v2/yarpcthrift/inbound_test.go

```diff
@@ -317,70 +335,29 @@ func TestUnaryHandlerBodyToWireError(t *testing.T) {
 		// XXX Body.ToWire returns error
 		return Response{Body: errorEnveloper{wire.Reply, fmt.Errorf("to wire error")}}, nil
 	}
-	h := thriftUnaryHandler{Protocol: proto, UnaryHandler: handler}
-	err := unaryCall(ctx, h)
+	h := UnaryHandler{Protocol: proto, ThriftHandler: handler}
+	res, resBody, err := unaryCall(ctx, h)
 
+	assert.Nil(t, res)
+	assert.Nil(t, resBody)
 	if assert.Error(t, err, "expected an error") {
 		assert.Contains(t, err.Error(), "to wire error")
 	}
 }
 
-func TestOnewayHandler(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	proto := thrifttest.NewMockEnvelopeAgnosticProtocol(mockCtrl)
-	// XXX expecting OneWay request instead of Call
-	proto.EXPECT().DecodeRequest(wire.OneWay, gomock.Any()).Return(
-		wire.NewValueStruct(wire.Struct{}), protocol.NoEnvelopeResponder, nil)
-
-	ctx, cancel := context.WithTimeout(context.Background(), testtime.Second)
-	defer cancel()
-
-	handler := func(ctx context.Context, v wire.Value) error {
-		return nil
-	}
-	h := thriftOnewayHandler{Protocol: proto, OnewayHandler: handler}
-	err := h.HandleOneway(ctx, request())
-
-	// XXX expecting success in this case
-	assert.NoError(t, err, "unexpected error")
-}
-
-func TestOnewayHandlerError(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	proto := thrifttest.NewMockEnvelopeAgnosticProtocol(mockCtrl)
-	// XXX mock returns decode request error, to induce error path out of handleRequest in HandleOneway
-	proto.EXPECT().DecodeRequest(wire.OneWay, gomock.Any()).Return(
-		wire.Value{}, protocol.NoEnvelopeResponder, fmt.Errorf("decode request error"))
-
-	ctx, cancel := context.WithTimeout(context.Background(), testtime.Second)
-	defer cancel()
-
-	handler := func(ctx context.Context, v wire.Value) error {
-		return nil
-	}
-	h := thriftOnewayHandler{Protocol: proto, OnewayHandler: handler}
-	err := h.HandleOneway(ctx, request())
-
-	if assert.Error(t, err, "expected an error") {
-		assert.Contains(t, err.Error(), "decode request error")
-	}
-}
-
-func request() *transport.Request {
-	return &transport.Request{
+func request() *yarpc.Request {
+	return &yarpc.Request{
 		Caller:    "caller",
 		Service:   "service",
 		Encoding:  "thrift",
 		Procedure: "MyService::someMethod",
-		Body:      bytes.NewReader([]byte("irrelevant")),
 	}
 }
 
-func unaryCall(ctx context.Context, h thriftUnaryHandler) error {
-	rw := new(transporttest.FakeResponseWriter)
-	return h.Handle(ctx, request(), rw)
+func requestBody() *yarpc.Buffer {
+	return yarpc.NewBufferBytes([]byte("irrelevant"))
+}
+
+func unaryCall(ctx context.Context, h UnaryHandler) (*yarpc.Response, *yarpc.Buffer, error) {
```

Comment:
optional: This seems a bit overkill as it just adds more indirection in the tests. Feel free to drop this test function if you think it will clean things up. Optional since this was from before and this PR is primarily a migration (not a clean up).